pkg/provider/aws/resources/eks.go

```diff
@@ -100,6 +104,21 @@ func CreateEksCluster(appName string, clusterName string, subnets []*Subnet, sec
 	}
 }
 
+// GetEksCluster will return the resource with the name corresponding to the appName and ClusterId
+//
+// If the dag does not contain the resource or the resource is not an EksCluster, it will return nil
+func GetEksCluster(appName string, clusterId string, dag *core.ResourceGraph) *EksCluster {
+	if clusterId == "" {
+		clusterId = DEFAULT_CLUSTER_NAME
+	}
+	cluster := NewEksCluster(appName, clusterId, nil, nil, nil)
+	resource := dag.GetResource(cluster.Id())
+	if existingCluster, ok := resource.(*EksCluster); ok {
+		return existingCluster
```

Comment:
When would `existingCluster` not be `cluster`?

Reply:
cluster wouldnt have the references to the subnets, or other things that the existing one would have, The only reason i create cluster is to be able to use it for its id

Reply:
Ah, right - I got confused between the `New` and `Create` functions.